File: backend/src/LinearPrecision.Api/Entities/Sprint.cs
using LinearPrecision.Shared.Domain;
using LinearPrecision.Shared.Domain.Enums;

namespace LinearPrecision.Api.Entities;

public class Sprint : TenantEntity, IAuditable
{
    public string Name { get; set; } = string.Empty;
    public string? Goal { get; set; }
    public Guid ProjectId { get; set; }
    public Project Project { get; set; } = null!;
    public SprintStatus Status { get; set; }
    public DateOnly StartDate { get; set; }
    public DateOnly EndDate { get; set; }
    public int? PlannedPoints { get; set; }
    public int? CompletedPoints { get; set; }

    // IAuditable
    public Guid? CreatedBy { get; set; }
    public Guid? UpdatedBy { get; set; }

    // Navigation properties
    public ICollection<TaskItem> Tasks { get; set; } = [];
}

File: lib/api/contracts/sprint.ts
export type SprintStatus = "Planned" | "Active" | "Completed" | "Cancelled" | number;

export interface SprintResponse {
  id: string;
  projectId: string;
  name: string;
  goal?: string | null;
  status: SprintStatus;
  startDate: string;
  endDate: string;
  plannedPoints?: number | null;
  completedPoints?: number | null;
  taskCount: number;
  completedTaskCount: number;
  createdAt: string;
  updatedAt: string;
}

export interface CreateSprintRequest {
  name: string;
  goal?: string | null;
  startDate: string;
  endDate: string;
}

export interface UpdateSprintRequest {
  name: string;
  goal?: string | null;
  startDate: string;
  endDate: string;
}

export interface SprintTransitionResponse {
  id: string;
  status: SprintStatus;
  completedPoints?: number | null;
}
